viam-cartographer_internal_test.go

```diff
@@ -243,6 +244,73 @@ func TestGetPositionEndpoint(t *testing.T) {
 	})
 }
 
+//nolint:dupl
+func TestGetPositionModularizationV2Endpoint(t *testing.T) {
+	svc := &cartographerService{Named: resource.NewName(slam.API, "test").AsNamed()}
+	mockCartoFacade := &cartofacade.Mock{}
+	svc.cartofacade = mockCartoFacade
+	svc.modularizationV2Enabled = true
+
+	var inputPose commonv1.Pose
+	var inputQuat map[string]interface{}
+
+	t.Run("origin pose success", func(t *testing.T) {
+		mockCartoFacade.GetPositionFunc = func(
+			ctx context.Context,
+			timeout time.Duration,
+		) (cartofacade.GetPosition, error) {
+			return cartofacade.GetPosition{
+					X:    0,
+					Y:    0,
+					Z:    0,
+					Real: 1.0,
+					Imag: 0.0,
+					Jmag: 0.0,
+					Kmag: 0.0,
+				},
+				nil
+		}
+
+		inputPose = commonv1.Pose{X: 0, Y: 0, Z: 0, OX: 0, OY: 0, OZ: 1, Theta: 0}
+		inputQuat = map[string]interface{}{"real": 1.0, "imag": 0.0, "jmag": 0.0, "kmag": 0.0}
+		pose, _, err := svc.GetPosition(context.Background())
```

Comment:
why are we not testing the component reference?

Reply:
I will add that back in, I got confused about using this after some of the changes from component reference to sensors[0], but this will still be needed after full mod in the case of multiple sensors, right?

Reply:
yes correct